server/common/utils/helpers.js

```diff
@@ -60,25 +61,34 @@ const responseWithList = (list, userId) => {
 };
 
 const responseWithListsMetaData = (lists, userId) =>
-  _map(lists, list => {
-    const { cohortId, favIds, items, ...rest } = list;
-    const activeItems = items.filter(item => !item.isArchived);
-    const doneItemsCount = activeItems.filter(item => item.isOrdered).length;
-    const unhandledItemsCount = activeItems.length - doneItemsCount;
-
-    const listToSend = {
-      ...rest,
-      doneItemsCount,
-      isFavourite: checkIfArrayContainsUserId(favIds, userId),
-      unhandledItemsCount
-    };
-
-    if (cohortId) {
-      listToSend.cohortId = cohortId;
-    }
-
-    return listToSend;
-  });
+  _filter(
+    _map(lists, list => {
+      const { cohortId: cohort, favIds, items, ...rest } = list;
+
+      if (cohort && cohort.isArchived === true) {
+        return;
+      }
+
+      const activeItems = items.filter(item => !item.isArchived);
+      const doneItemsCount = activeItems.filter(item => item.isOrdered).length;
+      const unhandledItemsCount = activeItems.length - doneItemsCount;
+      const cohortId = cohort ? cohort._id : null;
+
+      const listToSend = {
+        ...rest,
+        doneItemsCount,
+        isFavourite: checkIfArrayContainsUserId(favIds, userId),
+        unhandledItemsCount
+      };
+
+      if (cohortId) {
+        listToSend.cohortId = cohortId;
+      }
+
+      return listToSend;
+    }),
+    list => list
```

Comment:
why filter `list => list`?

Reply:
Because of lines 68-70: in _map() I check that if this is cohort's list and if this cohort is Archived. If so I exit current iteration. In the final array for all lists in archived cohorts, I get `undefined` so I use this strange filer to return only existing lists without these undefines.

Reply:
maybe use in this case https://lodash.com/docs/4.17.11#compact